viewing_party/party.py

```diff
@@ -1,23 +1,210 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    adds movie details to dictionary and returns movie details
+    if title, genre, or rating are None, function returns None
+    '''
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    if title is None or genre is None or rating is None:
+        return None
+    elif title == False or genre == False or rating == False:
+        return None
+    else:
+        return new_movie
+
+def add_to_watched(user_data, movie):
+    '''
+    adds movie data in the form of a dictionary
+    to a dictionary of user_data, where the key is watched and
+    the values are a list of dictionaries of movies the user has watched
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    adds movie data in the form of a dictionary
+    to a dictionary of user_data, where the key is watchlist and
+    the values are a list of dictionaries of movies the user has queued to watch
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    moves a movie from watchlist to watched
+    '''
+    for movie in user_data["watchlist"]:
+        if title in movie.values():
+            user_data["watchlist"].pop()
+            user_data["watched"].append(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    '''
+    calculates the average rating of the movies in the watched list
+    '''
+
+    total_movies_watched = len(user_data["watched"])
+    rating_total = 0
+
+    for movie in user_data["watched"]:
+        rating_total += movie["rating"]
+
+    if total_movies_watched == 0:
+        average_rating = 0
+    else:
+        average_rating = rating_total/total_movies_watched
+
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    '''
+    returns the genre most frequently listed in the watched list
+    '''
+    user_genre_count = {}
+
+    if len(user_data["watched"]) == 0:
+        most_watched_genre = None    
+    else:
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            
+            if genre not in user_genre_count.keys():
+                user_genre_count[genre] = 1
+            else:
+                user_genre_count[genre] += 1
+        
+        most_watched_genre = max(user_genre_count, key=user_genre_count.get)

```

Comment:
Nice use of `max`! For extra practice, reason through how you would do this with a normal for loop instead of using `max`.